drivers/kscan/kscan_xpt2046.c

```diff
@@ -0,0 +1,294 @@
+
+	if (!data->pressed || !data->enabled) {
+		return;
+	}
+
+	/* Check if touch is still pressed */
+	if (gpio_pin_get_dt(&config->int_gpio) == 0) {
+		data->pressed = false;
+		data->callback(data->dev, data->last_y, data->last_y, false);
+	} else {
+		/* Re-check later */
+		k_work_reschedule(&data->dwork, K_MSEC(10));
+	}
+}
+
+static void xpt2046_work_handler(struct k_work *kw)
+{
+	struct xpt2046_data *data = CONTAINER_OF(kw, struct xpt2046_data, work);
+	struct xpt2046_config *config = (struct xpt2046_config *)data->dev->config;
+
+	const struct spi_buf txb = {.buf = tbuf, .len = sizeof(tbuf)};
+	const struct spi_buf rxb = {.buf = data->rbuf, .len = sizeof(data->rbuf)};
+	const struct spi_buf_set tx_bufs = {.buffers = &txb, .count = 1};
+	const struct spi_buf_set rx_bufs = {.buffers = &rxb, .count = 1};
+
+	/* Run number of reads and calculate average */
+	int rounds = config->reads;
+	struct measurement meas = {0};
+
+	for (int i = 0; i < rounds; i++) {
+		if (xpt2046_read_and_cumulate(&config->bus, &tx_bufs, &rx_bufs, &meas) != 0) {
+			return;
+		}
+	}
+	meas.x /= rounds;
+	meas.y /= rounds;
+	meas.z /= rounds;
+
+	/* Calculate Xp = M * Xt + C using fixed point aritchmetics, where
+	 * Xp is the point in screen coordinates, Xt is the touch coordinates.
+	 * Use signed int32_t for calculation to ensure that we cover the roll-over to negative
+	 * values and return zero instead.
+	 */
+	int32_t mx = (config->screen_size_x << 16) / (config->max_x - config->min_x);
+	int32_t cx = (config->screen_size_x << 16) - mx * config->max_x;
+	int32_t x = mx * meas.x + cx;
+
+	x = (x < 0 ? 0 : x) >> 16;
+
+	int32_t my = (config->screen_size_y << 16) / (config->max_y - config->min_y);
+	int32_t cy = (config->screen_size_y << 16) - my * config->max_y;
+	int32_t y = my * meas.y + cy;
+
+	y = (y < 0 ? 0 : y) >> 16;
+
+	bool pressed = meas.z > config->threshold;
+
+	/* Don't send any other than "pressed" events.
+	 * releasing seem to cause just random noise
+	 */
+	if (data->enabled && pressed) {
+		LOG_DBG("raw: x=%4u y=%4u ==> x=%4d y=%4d", meas.x, meas.y, x, y);
+		data->last_x = x;
+		data->last_y = y;
+		data->pressed = pressed;
+		data->callback(data->dev, (uint32_t)y, (uint32_t)x, pressed);
+		/* Ensure that we send released event */
+		k_work_reschedule(&data->dwork, K_MSEC(100));
+	}
+	gpio_add_callback(config->int_gpio.port, &data->int_gpio_cb);
+}
+
+static int xpt2046_configure(const struct device *dev, kscan_callback_t callback)
+{
+	struct xpt2046_data *data = dev->data;
+
+	if (!callback) {
+		LOG_ERR("Callback is null");
+		return -EINVAL;
+	}
+	LOG_DBG("%s: set callback", dev->name);
+
+	data->callback = callback;
+
+	return 0;
+}
+
+static int xpt2046_enable_callback(const struct device *dev)
+{
+	struct xpt2046_data *data = dev->data;
+	const struct xpt2046_config *config = dev->config;
+
+	LOG_DBG("%s: enable cb", dev->name);
+	data->enabled = true;
+	gpio_add_callback(config->int_gpio.port, &data->int_gpio_cb);
+	return 0;
```

Comment:
Unfortunately, I've not noticed that one before. Would you mind fixing that by adding an `nl` to all places where it is missing?  Otherwise LGTM!

Reply:
Sure.
Thanks for being pedantic.